### ISSUE TITLE
Fails to parse TTF file -> "Error while parsing file: requested 1 bytes, but only 0 bytes available"

### ISSUE THREAD

**Issue by N00BIER** [Dvd848/pytai#8]

**Comment by Dvd848** [comment]
Thanks for reporting this!

I tried a few `ttf` files but couldn't reproduce the exact error.
I was able to reproduce two other errors instead:

- [Roboto](https://fonts.google.com/specimen/Roboto) ->`"'utf-16-be' codec can't decode byte 0x68 in position 4: truncated data"`
- Courier New -> `"'ascii' codec can't decode byte 0xa9 in position 1: ordinal not in range(128)"`

From a quick look, both of these errors seem related to the way the core Kaitai library itself handles the files. I'll double check this again later this week. For what it's worth, I was able to bypass the errors by applying the following patch:

```
diff --git a/pytai/kaitai/formats/ttf.py b/pytai/kaitai/formats/ttf.py
index dad7930..9b97063 100644
--- a/pytai/kaitai/formats/ttf.py
+++ b/pytai/kaitai/formats/ttf.py
@@ -276,7 +276,7 @@ class Ttf(KaitaiStruct):
                 _pos = io.pos()
                 io.seek((self._parent.ofs_strings + self.ofs_str))
                 self._debug['_m_ascii_value']['start'] = io.pos()
-                self._m_ascii_value = (io.read_bytes(self.len_str)).decode(u"ascii")
+                self._m_ascii_value = (io.read_bytes(self.len_str)).decode(u"ascii", errors="ignore")
                 self._debug['_m_ascii_value']['end'] = io.pos()
                 io.seek(_pos)
                 return getattr(self, '_m_ascii_value', None)
@@ -290,7 +290,7 @@ class Ttf(KaitaiStruct):
                 _pos = io.pos()
                 io.seek((self._parent.ofs_strings + self.ofs_str))
                 self._debug['_m_unicode_value']['start'] = io.pos()
-                self._m_unicode_value = (io.read_bytes(self.len_str)).decode(u"utf-16be")
+                self._m_unicode_value = (io.read_bytes(self.len_str)).decode(u"utf-16be", errors="ignore")
                 self._debug['_m_unicode_value']['end'] = io.pos()
                 io.seek(_pos)
                 return getattr(self, '_m_unicode_value', None)
```

1. Does the workaround above work for your file as well?
2. If not, would it be possible for you to attach the problematic `ttf` file?

Thanks!

**Comment by N00BIER** [comment]
Thanks for a quick turnaround!

> * [Roboto](https://fonts.google.com/specimen/Roboto) ->`"'utf-16-be' codec can't decode byte 0x68 in position 4: truncated data"`
> * Courier New -> `"'ascii' codec can't decode byte 0xa9 in position 1: ordinal not in range(128)"`

I've had those too.

Have you pushed the update to the python repo, so I could upgrade it with pip --upgrade?

**Comment by Dvd848** [comment]
Ideally I'd like to make sure that we fix all known TTF parsing issues before releasing a new official release.
Would you be able to test the fix on your font by running the following (or alternatively attach it here)?

```console
git clone https://github.com/Dvd848/pytai.git pytai_test && cd pytai_test/
sed -i -e 's/decode(u"\([^"]*\)")/decode(u"\1", errors="ignore")/g' pytai/kaitai/formats/ttf.py
python3 __main__.py /path/to/your/font.ttf -kf ttf
```

Be sure to replace `/path/to/your/font.ttf` with the correct path.
Thanks!

**Comment by N00BIER** [comment]
apologies for delay. I'll check over the weekend and revert.

**Comment by N00BIER** [comment]
> Ideally I'd like to make sure that we fix all known TTF parsing issues before releasing a new official release. Would you be able to test the fix on your font by running the following (or alternatively attach it here)?
> 
> ```
> git clone https://github.com/Dvd848/pytai.git pytai_test && cd pytai_test/
> sed -i -e 's/decode(u"\([^"]*\)")/decode(u"\1", errors="ignore")/g' pytai/kaitai/formats/ttf.py
> python3 __main__.py /path/to/your/font.ttf -kf ttf
> ```
> 
> Be sure to replace `/path/to/your/font.ttf` with the correct path. Thanks!

Now, seems to be working fine. Thanks a lot for your effort!

**Comment by Dvd848** [comment]
Thanks for your feedback!
After diving a bit more into the issue, I chose to apply a different workaround.
I've already submitted the workaround to the repository, and it can be tested with:

```
git clone https://github.com/Dvd848/pytai.git pytai_test && cd pytai_test/
python3 __main__.py /path/to/your/font.ttf -kf ttf
```

I'll hopefully be able to release a new official version to PyPI some time later this week. 
If you want, feel free to check your font with the latest code and let me know if you encounter any issues with the revised workaround.
